### PR TITLE
Add coverage for the daemon_utils module.

### PR DESCRIPTION
**Pull request by timuralp**
Adds tests for setting up the logger and daemon context.

### CODE REVIEW

File: test/unit/test_daemon_utils.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            del sys.modules['swift']
+
+    def tearDown(self):
+        if 'swift' in sys.modules:
+            del sys.modules['swift']
+        if self.old_swift_module:
+            sys.modules['swift'] = self.old_swift_module
+
+    @mock.patch('s3_sync.daemon_utils.logging')
+    def test_old_swift_raises_exception(self, loggin_mock):
```
Comment (dbishop) [inline test/unit/test_daemon_utils.py]: 2nd arg probably supposed to be `logging_mock`?

File: test/unit/test_daemon_utils.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        import_mock.side_effect = _import
+        with self.assertRaises(ImportError):
+            s3_sync.daemon_utils.load_swift('logger', once=True)
+        import_mock.assert_called_once_with('swift', mock.ANY, None, None)
+
+    @mock.patch('sys.stdout', new_callable=StringIO.StringIO)
+    @mock.patch('s3_sync.daemon_utils.sys')
+    @mock.patch('s3_sync.daemon_utils.os.path.exists')
+    def test_setup_context_exits(self, exists_mock, sys_mock, mock_stdout):
```
Comment (dbishop) [inline test/unit/test_daemon_utils.py]: I tend to prefer `mock_*` for mock var names, but as long as they're consistent w/in a small-ish scope, it doesn't matter.  But here, the 4th arg should probably be `stdout_mock` for consistency.

File: test/unit/test_daemon_utils.py (hunk 1 above)
Comment (timuralp) [reply test/unit/test_daemon_utils.py]: Done

File: test/unit/test_daemon_utils.py (hunk 2 above)
Comment (timuralp) [reply test/unit/test_daemon_utils.py]: Done

**Pull request merged**